esx_service/vmci/vmci_server.c

```diff
@@ -68,18 +70,17 @@ vmci_init(void)
 
    af = vsock_get_family();
    if (af == -1) {
-      fprintf(stderr, "Failed to get address family.\n");
-      return CONN_FAILED_VMCI_ADDRESS_FAMILY_GET;
+      return CONN_FAILURE;
    }
 
    /*
     * Open a STREAM socket using our address family.
     */
 
-   s = socket(af, SOCK_STREAM, 0);
-   if (s == -1) {
+   socket_fd = socket(af, SOCK_STREAM, 0);
+   if (socket_fd == -1) {
       perror("Failed to open socket");
```

Comment:
Do you need this perror? What if perror runs into issues in opening STDERR and resets errno? Also errno valid after perror returns successfully?